src/test/java/io/vertx/test/core/Http1xTest.java

```diff
@@ -3600,6 +3618,69 @@ private void testPerPeerPooling(Function<Integer, HttpClientRequest> requestProv
   }
 
   @Test
+  public void testHeadMDontAutomaticallySetContentHeaders() throws Exception {
+    testHeadMustNoAutomaticallySetContentHeaders(MultiMap.caseInsensitiveMultiMap(), respHeaders -> {
+      assertFalse(respHeaders.contains("Content-Length"));
+      assertFalse(respHeaders.contains("Transfer-Encoding"));
+    });
+  }
+
+  @Test
+  public void testHeadMustNotSendBodyWhenContentLengthSet() throws Exception {
+    MultiMap reqHeaders = MultiMap.caseInsensitiveMultiMap();
+    reqHeaders.set("Content-Length", "10");
+    testHeadMustNoAutomaticallySetContentHeaders(reqHeaders, respHeaders -> {
+      assertEquals(respHeaders.get("Content-Length"), " 10");
+      assertFalse(respHeaders.contains("Transfer-Encoding"));
+    });
+  }
+
+  @Ignore("See https://github.com/netty/netty/issues/6761")
+  @Test
+  public void testHeadMustNotSendBodyWhenTransferEncodingSet() throws Exception {
+    MultiMap reqHeaders = MultiMap.caseInsensitiveMultiMap();
+    reqHeaders.set("Transfer-Encoding", "chunked");
+    testHeadMustNoAutomaticallySetContentHeaders(reqHeaders, respHeaders -> {
+      assertEquals(respHeaders.get("Content-Length"), "10");
+      assertFalse(respHeaders.contains("Transfer-Encoding"));
+    });
+  }
+
+  private void testHeadMustNoAutomaticallySetContentHeaders(MultiMap reqHeaders, Consumer<MultiMap> headersChecker) throws Exception {
+    server.requestHandler(req -> {
+      HttpServerResponse resp = req.response();
+      resp.headers().addAll(reqHeaders);
+      resp.end();
+    });
+    startServer();
+    NetClient client = vertx.createNetClient();
```

Comment:
Why do we need to use a net client? Because the http client would automatically add some headers?